include/seqan3/range/view/translation.hpp

```diff
@@ -45,292 +45,657 @@
     }
 
-    /*!\brief Pipe operator that enables view-typical use of pipe notation.
-    * \tparam irng_t The type of the range being processed.
-    * \param irange The range being processed as left argument of the pipe.
-    * \param translate_single_fn_v The result of the single-argument operator().
-    */
-    template <typename irng_t>
-    friend auto operator|(irng_t && irange, seqan3::detail::translate_single_fn const & translate_single_fn_v)
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Return the n-th element.
+     * \param n The element to retrieve.
+     *
+     * Accessing an element behind the last causes undefined behaviour. In debug mode an assertion checks the size of
+     * the container.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (never modifies data).
+     */
+    reference operator[](size_type const n) const
+    {
+        assert(n < size());
+        switch (data_members->tf)
+        {
+            case translation_frames::FWD_FRAME_0:
+                return translate_triplet((data_members->urange)[n * 3], (data_members->urange)[n * 3 + 1], (data_members->urange)[n * 3 + 2]);
+                break;
+            case translation_frames::REV_FRAME_0:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 1]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 2]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]));
+                break;
+            case translation_frames::FWD_FRAME_1:
+                return translate_triplet((data_members->urange)[n * 3 + 1], (data_members->urange)[n * 3 + 2], (data_members->urange)[n * 3 + 3]);
+                break;
+            case translation_frames::REV_FRAME_1:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 2]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 4]));
+                break;
+            case translation_frames::FWD_FRAME_2:
+                return translate_triplet((data_members->urange)[n * 3 + 2], (data_members->urange)[n * 3 + 3], (data_members->urange)[n * 3 + 4]);
+                break;
+            case translation_frames::REV_FRAME_2:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 4]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 5]));
+                break;
+            default:
+                throw std::invalid_argument(multiple_frame_error);
+                break;
+        }
+    }
+    //!\}
+
+    //!\brief Implicit conversion to container types.
+    template <random_access_sequence_concept container_type>
+    operator container_type()
```

Comment:
I think it would be better if the conversion were marked `explicit` so that it only happens when someone calls `std::vector<aa27>{v}` and not by accident, because some function expects a vector, you don't know and it suddenly starts copying...

Reply:
see also http://en.cppreference.com/w/cpp/language/cast_operator